.jenkins/nightly.groovy

```diff
@@ -17,7 +17,7 @@ pipeline {
                     }
                     environment {
                         CTEST_OPTIONS = '--timeout 180 --no-compress-output -T Test'
-                        CMAKE_OPTIONS = '-D CMAKE_BUILD_TYPE=Release -D CMAKE_CXX_STANDARD=17 -D CMAKE_CXX_EXTENSIONS=OFF'
```

Comment:
Why did you remove?  Relying on defaults rather being explicit about it?

Reply:
My assumption was that it was not so much relying on the exact standard, rather it was requiring at least C++17. As this PR now provides C++17, it seemed no longer necessary.

Reply:
This variable was enforcing consistent options on Kokkos and ArborX.  I actually think it is a good idea.  Assume we of a C++20 build for instance.

Reply:
I restored it back.